src/reducers/computeRenderedRows.js

```diff
@@ -207,35 +211,68 @@ function computeRenderedRowOffsets(state, rowRange) {
     return;
   }
 
-  const bufferMapping = []; // state.rows
+  // output for this function
+  const bufferMapping = state.rows;
   const rowOffsetsCache = {}; // state.rowHeights
```

Comment:
rowOffsetsCache is going to have endBufferIdx-firstBufferIdx items in it.  Is there any risk of that being less than the number of items in bufferMapping if we have a larger number of rows in the buffer on the previous render?  Has any work been done to avoid rows rendered in the buffer changing by +/-1 as we scroll?

Reply:
> This is fixed by keeping the old rowOffsets in the rowsNotInBuffer.forEach call below.

Unfortunately, there's a bug here. This fix is only done if `scrolling` is being done. 
So to fix the fix, I defaulted back to the old behavior of having a fresh empty array of `bufferMapping` while not `scrolling`. Basically this means that the entire logic of keeping the old rows + offsets are only done while scrolling (exactly as previous behavior without buffering). 
This issue must have crept in after the refactor where I merged the 2 `computeRenderedRowOffsets` functions)

Reply:
I'm concerned about rowOffsets & the mapping being allowed to drift.  It violates certain expectations of how the code is written and may make it difficult to maintain.  Could you modify the change to keep the data structures in sync so there are reasonable invariants about how they can be used together?  

If there is a need to handle rows in the viewport and rows outside of the viewport separately, let's split those into 2 data structures to create cleaner code.  We can use a selector to merge those data structures where needed.